apng/__init__.py

```diff
@@ -30,12 +30,14 @@ def is_png(png):
 	"""
 	if isinstance(png, str):
 		with open(png, "rb") as f:
-			png = f.read(8)
+			png_header = f.read(8)
 			
 	if hasattr(png, "read"):
-		png = png.read(8)
+		position = png.tell()
+		png_header = png.read(8)
+		png.seek(position)
 		
-	return png[:8] == PNG_SIGN
+	return png_header[:8] == PNG_SIGN
```

Comment:
Good catch. However bytes object is not handled correctly. Please use something like:
```py
if isinstance(png, str):
	...
elif hasattr(png, "read"):
	...
elif isinstance(png, bytes):
	png_header = png[:8]
else:
	raise TypeError("Muse be file, bytes, or str")
```
Also no need to slice it again:
```py
return png_header == PNG_SIGN
```

Reply:
Done.